src/flows/pipes/Schedule/ExportTaskOverlay/context.tsx

```diff
@@ -0,0 +1,209 @@
+// Libraries
+import React, {FC, useState, useCallback, useEffect, useContext} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {parse, format_from_js_file} from '@influxdata/flux'
+
+// Components
+import {
+  resourceLimitReached,
+  taskCreatedSuccess,
+  taskNotCreated,
+  taskUpdateSuccess,
+  taskUpdateFailed,
+} from 'src/shared/copy/notifications'
+import {ASSET_LIMIT_ERROR_STATUS} from 'src/cloud/constants/index'
+import {postTask, patchTask} from 'src/client'
+
+// Types
+import {Task} from 'src/types'
+
+// Actions
+import {getTasks} from 'src/tasks/actions/thunks'
+import {notify} from 'src/shared/actions/notifications'
+import {checkTaskLimits} from 'src/cloud/actions/limits'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {getErrorMessage} from 'src/utils/api'
+import {getOrg} from 'src/organizations/selectors'
+import {remove} from 'src/flows/context/query'
+
+// Contexts
+import {PopupContext} from 'src/flows/context/popup'
+
+export enum ExportAsTask {
+  Create = 'create',
+  Update = 'update',
+}
+
+interface ContextType {
+  script: string
+  submit: () => void
+  activeTab: ExportAsTask
+  handleSetActiveTab: (tab: ExportAsTask) => void
+  selectedTask: Task | undefined
+  handleSetTask: (task: Task) => void
+  taskName: string
+  handleSetTaskName: (name: string) => void
+}
+
+const DEFAULT_CONTEXT: ContextType = {
+  script: '',
+  submit: () => {},
+  activeTab: ExportAsTask.Create,
+  handleSetActiveTab: _ => {},
+  selectedTask: undefined,
+  handleSetTask: _ => {},
+  taskName: '',
+  handleSetTaskName: _ => {},
+}
+
+export const Context = React.createContext<ContextType>(DEFAULT_CONTEXT)
+
+export const Provider: FC = ({children}) => {
+  const dispatch = useDispatch()
+  const org = useSelector(getOrg)
+  const [activeTab, setActiveTab] = useState(ExportAsTask.Create)
+  const [selectedTask, setSelectedTask] = useState<Task>(undefined)
```

Comment:
would we want to set a default here when retrieving all the tasks?

Reply:
it didn't seem to add much to the equation to implicitly update the first task you have instead of instructing the user to "select a task first". the former seems potentially destructive? even on an user interaction as innocuous as double clicking. Is your concern about product consistency?

Reply:
I guess so. No concern, just thinking